linear_algebra/src/polynom_for_points.py

```diff
@@ -24,96 +24,79 @@ def points_to_polynomial(coordinates: list[list[int]]) -> str:
     >>> print(points_to_polynomial([[1, 5], [2, 2], [3, 9]]))
     f(x)=x^2*5.0+x^1*-18.0+x^0*18.0
     """
-    try:
-        check = 1
-        more_check = 0
-        d = coordinates[0][0]
-        for j in range(len(coordinates)):
-            if j == 0:
-                continue
-            if d == coordinates[j][0]:
-                more_check += 1
-                solved = "x=" + str(coordinates[j][0])
-                if more_check == len(coordinates) - 1:
-                    check = 2
-                    break
-                elif more_check > 0 and more_check != len(coordinates) - 1:
-                    check = 3
-                else:
-                    check = 1
+    if len(coordinates) == 0 or not all(len(pair) == 2 for pair in coordinates):
+        return "The program cannot work out a fitting polynomial."
+
+    if len({tuple(pair) for pair in coordinates}) != len(coordinates):
+        return "The program cannot work out a fitting polynomial."
 
-        if len(coordinates) == 1 and coordinates[0][0] == 0:
-            check = 2
-            solved = "x=0"
-    except Exception:
-        check = 3
+    set_x = {x for x, _ in coordinates}
+    if len(set_x) == 1:
+        return "x=" + str(coordinates[0][0])
```

Comment:
Shorter, faster, and easier to read.
```suggestion
        return f"x={coordinates[0][0]}"
```